### DISCUSSION TITLE
Will stbrp_pack_rects reorder the rects array?

### DISCUSSION THREAD

**Discussion by zwcloud** [nothings/stb#1232] [Q&A] [answered]
In the documentation

> as the function temporarily reorders the array while it executes.

But when `stbrp_pack_rects` is finished, will the array be restored to the original order?

I want to know this because if the original order is kept, there will be no need to find a packed rect according to its `id`.

**Comment by nothings** [comment, marked as answer]
yes, the original order is restored, that's what "temporarily" is meant to imply
